daffodil-lib/src/main/scala/org/apache/daffodil/util/DecimalUtils.scala

```diff
@@ -335,4 +335,142 @@ object DecimalUtils {
     outArray
   }
 
+  def convertFromAsciiStandard(digit: Int): (Int, Boolean) = {
+    if ((digit >= 48) && (digit <= 57)) // positive 0-9
+      return (digit - 48, false)
+    else if ((digit >= 112) && (digit <= 121)) // negative 0-9
+      return (digit - 112, true)
+    else
+      throw new NumberFormatException("Invalid zoned digit: " + digit)
+  }
```

Comment:
I'll add some javadoc to this.
For AsciiStandard 0-9 represents overpunched positive 0-9 and p-y represent overpunched negative 0-9.  The boolean represents whether or not the digit was an overpunched negative. This is needed to handle the possibility of the signed digit being a 0.

For example, 123p represents -1230 in AsciiStandard, but the signed (overpunched) digit is a 0.  With the other digits I had originally just returned a negative version of them, but -0 == 0, so I needed some way to convey the fact that we are dealing with a negative number back to the where this function was called.

I will update the conversion functions to take chars instead of ints, as that will be more clear.